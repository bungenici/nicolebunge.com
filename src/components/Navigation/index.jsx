import classNames from 'classnames';
import { Link } from 'gatsby';
import PropTypes from 'prop-types';
import React from 'react';
import styles from './style.module.css';

const Navigation = (props) => {
  const { className, inline, routes, ...otherProps } = props;

  return (
    <nav className={classNames(className, styles.navigation)} {...otherProps}>
      {routes && (
        <ul
          className={classNames(styles.navigation__list, {
            [styles.navigation__listInline]: inline,
          })}
        >
          {routes.map((route) => (
            <li className={styles.navigation__item} key={route.path}>
              <Link
                to={route.path}
                className={styles.navigation__link}
                activeClassName={styles.navigation__linkActive}
              >
                {route.name}
              </Link>
            </li>
          ))}
        </ul>
      )}
    </nav>
  );
};

Navigation.propTypes = {
  className: PropTypes.string,
  inline: PropTypes.bool,
  routes: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      path: PropTypes.string,
    }),
  ),
};

Navigation.defaultProps = {
  className: '',
  inline: false,
  routes: [],
};

export default Navigation;
