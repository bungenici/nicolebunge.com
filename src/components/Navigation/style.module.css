.navigation {
  color: var(--color-primary);
  font-weight: 600;
}

.navigation__list {
  display: flex;
  flex-wrap: wrap;
  list-style: none;
  margin: -1em;
  padding: 0;
}

.navigation__item {
  margin: 1em;
}

.navigation__link {
  color: #333;
  display: inline-block;
  padding-bottom: 1.5em;
  padding-top: 1.5em;
  position: relative;
  transition: color 0.3s;
}

.navigation__link--active {
  color: var(--color-primary);
}

.navigation__link--active::after {
  background-color: var(--color-primary);
  content: '';
  height: 0.25em;
  position: absolute;
  right: 0;
  top: 0;
  width: 100%;
}
