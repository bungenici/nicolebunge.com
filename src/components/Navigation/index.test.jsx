import React from 'react';
import renderer from 'react-test-renderer';
import Navigation from '.';

describe('Navigation component', () => {
  it('renders correctly', () => {
    const routes = [
      {
        name: 'Index',
        path: '/',
      },
    ];
    const tree = renderer.create(<Navigation routes={routes} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
