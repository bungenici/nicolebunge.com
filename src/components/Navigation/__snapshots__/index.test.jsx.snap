// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component renders correctly 1`] = `
<nav
  className="navigation"
>
  <ul
    className="navigation__list"
  >
    <li
      className="navigation__item"
    >
      <a
        activeClassName="navigation__linkActive"
        className="navigation__link"
        href="/"
      >
        Index
      </a>
    </li>
  </ul>
</nav>
`;
